usaspending_api/recipient/delta_models/recipient_lookup.py

```diff
@@ -1,30 +1,552 @@
+            AND temp_collect_recipients_view.uei IS NULL
+        )
+        WHERE
+            -- Limit to the "temp_collect_recipients_view" records that do not JOIN to another record
+            uei_and_duns_recipients.duns_recipient_hash IS NULL
+    )
+    """,
+    # -----
+    # Update the temporary_restock_recipient_lookup table to include any alternate names
+    # -----
+    fr"""
+    CREATE OR REPLACE TEMPORARY VIEW temp_recipients_with_alt_names_view AS (
+        WITH recipient_names (
+            WITH alt_names AS (
+                SELECT
+                    recipient_hash,
+                    COLLECT_SET(awardee_or_recipient_legal) AS all_names
+                FROM temp_transaction_recipients_view
+                WHERE COALESCE(awardee_or_recipient_legal, '') != ''
+                GROUP BY recipient_hash
+            ),
+            alt_parent_names AS (
+                SELECT
+                    parent_recipient_hash AS recipient_hash,
+                    COLLECT_SET(ultimate_parent_legal_enti) AS all_names
+                FROM temp_transaction_recipients_view
+                WHERE COALESCE(ultimate_parent_legal_enti, '') != ''
+                GROUP BY parent_recipient_hash
+            )
+            SELECT
+                COALESCE(an.recipient_hash, apn.recipient_hash) AS recipient_hash,
+                COALESCE(
+                    ARRAY_SORT(ARRAY_UNION(COALESCE(an.all_names, ARRAY()), COALESCE(apn.all_names, ARRAY()))),
+                    ARRAY()
+                 ) AS all_names
+            FROM alt_names AS an
+            FULL OUTER JOIN alt_parent_names AS apn ON an.recipient_hash = apn.recipient_hash
+        )
+        SELECT
+            {
+                ", ".join(
+                    [
+                        "temp_stale_recipients_removed_view." + val
+                        for val
+                        in set(RPT_RECIPIENT_LOOKUP_DELTA_COLUMNS) - {"alternate_names"}
+                    ]
+                )
+            },
+            COALESCE(
+                ARRAY_REMOVE(recipient_names.all_names, COALESCE(temp_stale_recipients_removed_view.legal_business_name, '')),
+                ARRAY()
+            ) AS alternate_names
+        FROM temp_stale_recipients_removed_view
+        LEFT OUTER JOIN recipient_names ON (
+            temp_stale_recipients_removed_view.recipient_hash = recipient_names.recipient_hash
+        )
+    )
+    """,
+    # -----
+    # Insert the temporary_restock_recipient_lookup table into recipient_lookup
+    # -----
+    fr"""
+    INSERT OVERWRITE {{DESTINATION_DATABASE}}.{{DESTINATION_TABLE}}
+    (
+        {",".join([col for col in RECIPIENT_LOOKUP_COLUMNS_WITHOUT_ID])}
+    )
+    SELECT
+        {",".join([col for col in RECIPIENT_LOOKUP_COLUMNS_WITHOUT_ID])}
+    FROM
+        temp_recipients_with_alt_names_view
+    """,
+    # -----
+    # Cleanup the temporary table and views
```

Comment:
Minor thing here to remove "table" from the comment, as I think all that's left is views.